docs/resource-ids.md

```diff
@@ -0,0 +1,51 @@
+# Resource IDs
+
+Every Pulumi resource must have an `"id"` field. `"id"` must be a string, and it must be
+set by the provider (`Computed` and *not* `Optional` in Terraform parlance).
+
+## SDKv1 and SDKv2 Based Providers
+
+The ID requirement is easy to satisfy for SDKv{1,2} based providers since both SDKs
+require an ID field of type string set on the provider. If the provider being bridged is
+based on SDKv1 or SDKv2, then the bridge handles Pulumi's ID field without intervention.
+
+## PF Based Providers
+
+Most PF based providers have an attribute called `"id"` of the right kind for the bridge
+to use. If your provider doesn't, then you will see an error when `make tfgen` is
+run. Each error has a different kind of resolution.
+
+### ID of the wrong type
+```
+error: Resource dnsimple_email_forward has a problem: "id" attribute is of type "int", expected type "string". To map this resource consider overriding the SchemaInfo.Type field or specifying ResourceInfo.ComputeID.
+```
+
+This error happens when the upstream resource has an ID but it's not a string. You can fix
+it by setting the `SchemaInfo.Type` override for the `"id"` field:
+
+```go
+			"dnsimple_email_forward": {
+				Fields: map[string]*tfbridge.SchemaInfo{
+					"id": {Type: "string"},
```

Comment:
This only works if the actual type can be coalesced to string right? Like it's a number. It is not an object.

Reply:
Yes. I'll add a note explaining that.